test/Utility/CMakeLists.txt

```diff
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 3.20.4 FATAL_ERROR)
+
+include_directories("${CMAKE_SOURCE_DIR}/test")
+
+set(TEST_UTILITY_SOURCES IndexedArray.cpp)
+
+add_library(test_utility STATIC OBJECT ${TEST_UTILITY_SOURCES})
+target_compile_definitions(test_utility PRIVATE TESTGROUP=Utility)
```

Comment:
I'll need to test how this interoperates with IDE integration, will report back.

Reply:
Works fine.

Still I'd prefer not to introduce any additional macros that require externally-passed definitions to work.

Reply:
Please elaborate what is your concern in that case? I thought it will be useful and it was added merely for convenience to have a unified way to group tests. If concern is only about potentially missing passed define then it will still compile fine, just won't be pretty. We can additionally guard it with #ifdef if you like, e.g.:
```
#ifdef TESTGROUP
#define GTEST(TestBase, TestName) TEST(TESTGROUP, TestBase##_##TestName)
#else
#define GTEST TEST
#endif
```

If you are against passing define from cmake and/or using macro at all then also explain why...

Reply:
Not all IDEs handle those well, that's the main reason.

Solution with the #ifdef looks OK.

Reply:
Not all IDEs handle those well, that's the main reason.

Solution with the #ifdef looks OK.